File: src/main/java/com/example/yourssu2/model/Memo.java
package com.example.yourssu2.model;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.Column;
import javax.persistence.EntityListeners;
import javax.persistence.MappedSuperclass;
import java.time.LocalDateTime;

@Getter
@Setter
@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public class Memo {
    private int id;
    private String title;
    private String text;

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createdDate;

    @LastModifiedDate
    private LocalDateTime modifiedDate;

    @Builder
    public Memo() {}

    @Builder
    public Memo(int id, String title, String text) {
        this.id = id;
        this.title = title;
        this.text = text;
        this.createdDate = LocalDateTime.now();
        this.modifiedDate = LocalDateTime.now();
    }

}

File: src/main/java/com/example/yourssu2/model/Memo.kt
package com.example.yourssu2.model

import com.fasterxml.jackson.annotation.JsonFilter
import lombok.Getter
import lombok.Setter
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import javax.persistence.*


@Getter
@Setter
@MappedSuperclass
@JsonFilter("MemoFilter")
@EntityListeners(AuditingEntityListener::class)
open class Memo(
    @Id
    @GeneratedValue
    var id: Int,
    var title: String?,
    var text: String?,
    var createdAt: String?,
    var updatedAt: String?
)